sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -196,6 +196,8 @@ trait ColumnResolutionHelper extends Logging with DataTypeErrorsBase {
             u.copy(child = newChild)
           }
 
+        case s : ScopedExpression => s.mapScope(innerResolve(_, isTopLevel = false))
```

Comment:
so we may resolve the `UnresolvedAttribute` inside `ScopedExpression` using the `resolveColumnByName` (which is usually the child plan output), instead of the scoped attributes?